gobblin-compaction/src/main/java/gobblin/compaction/hivebasedconstructs/MRCompactionEntity.java

```diff
@@ -0,0 +1,23 @@
+package gobblin.compaction.hivebasedconstructs;
+
+import java.util.Properties;
+import lombok.Getter;
+
+
+/**
+ * Entity that stores information required for launching an {@link gobblin.compaction.mapreduce.MRCompactor} job
+ */
+@Getter
+public class MRCompactionEntity {
+  private final String primaryKey;
```

Comment:
Besides Chavdar's comment, if `primaryKey` is present,  do you need to set `COMPACTION_JOB_DEDUP_KEY`? Otherwise, the compactor will use the default `ALL` option for dedupe.